java/ql/test/library-tests/frameworks/play/resources/Resource.java

```diff
@@ -0,0 +1,37 @@
+import play.mvc.Controller;
+import play.mvc.Http.*;
+import play.mvc.Results;
+import play.mvc.Result;
+import play.filters.csrf.AddCSRFToken;
+import play.mvc.BodyParser;
+import play.libs.F;
+import java.util.concurrent.CompletionStage;
+
+
+public class Resource extends Controller {
+
+    @AddCSRFToken
+    public Result index() {
+        response().setHeader("X-Play-QL", "1");
+        return ok("It works!");
+    }
+
+    @BodyParser.Of()
+    public Result session_redirect_me(String uri) {
+        String url = request().getQueryString("url");
+        redirect(url);
+      }
+
+    public F.Promise<Result> async_promise(String token) {
+        ok(token);
+      }
+
+    public CompletionStage<Result> async_completionstage(String uri) {
+        ok("Async completion Stage");
+      }
```

Comment:
These methods don't look like they compile, since they're missing `return` statements, but I can see that the test somehow runs - I'll need to dig a bit deeper here to figure out what's going on.

Reply:
@aschackmull is it the test stub code for functions like `ok`, `redirect`, you are referring to, that should return?. Let me know if any changes are required on my side.

Reply:
It looks like there's a bug in our tooling - the tests ought to have failed with a compilation error, since these three methods don't have return statements, but somehow it accepted it. That's on us and we'll look into fixing it. We don't need to wait for a fix in the tooling, but could you please add `return` statements to these three methods? I presume that the first should return the result of the call to `redirect`, but I'm not sure what the other two are meant to return.

Reply:
(The test stub code should also have a bunch of `return null;`, but we can fix that when we fix the tooling bug.)

Reply:
@aschackmull updated with return statements for the methods.